### PR TITLE
Add `get_city_name()` method to dataloader, to fetch name of city where a log was captured.

### PR DESCRIPTION
**Pull request by johnlambert-argo**
## PR Summary
Add `get_city_name()` method to dataloader, to fetch name of city where a log was captured. Works for TbV, Sensor, or LIDAR datasets.

## Testing

In order to ensure this PR works as intended, it is:

* [ ] unit tested.
* [ ] other or not applicable (*additional detail/rationale required*)

## Compliance with Standards

As the author, I certify that this PR conforms to the following standards:

* [ ] Code changes conform to [PEP8](https://www.python.org/dev/peps/pep-0008) and docstrings conform to the Google Python [style guide](https://google.github.io/styleguide/pyguide.html?showone=Comments#38-comments-and-docstrings).
* [ ] A well-written summary explains what was done and why it was done.
* [ ] The PR is adequately tested and the testing details and links to external results are included.

### CODE REVIEW

File: src/av2/datasets/sensor/av2_sensor_dataloader.py [hunk 1]
```diff
@@ -136,6 +132,28 @@ def get_log_map_dirpath(self, log_id: str) -> Path:
...
+            log_id: unique ID of vehicle log.
+
+        Returns:
+            Name of the city where the log of interest was captured.
+
+        Raises:
+            RuntimeError: If no vector map file is found for the query log ID.
+        """
+        vector_map_fpaths = list(self._data_dir.glob(f"{log_id}/map/log_map_archive*"))
+        if len(vector_map_fpaths) != 1:
```
Comment (benjaminrwilson) [inline src/av2/datasets/sensor/av2_sensor_dataloader.py:151]: Could we add a check for if there's an extra vector map file in the directory?

File: src/av2/datasets/sensor/av2_sensor_dataloader.py [hunk 2]
```diff
@@ -136,6 +132,28 @@ def get_log_map_dirpath(self, log_id: str) -> Path:
...
+            raise RuntimeError(f"Vector map file is missing for {log_id}")
```
Comment (benjaminrwilson) [inline src/av2/datasets/sensor/av2_sensor_dataloader.py:152]:
```suggestion
            raise RuntimeError(f"Vector map file is missing for {log_id}.")
```

File: src/av2/datasets/sensor/av2_sensor_dataloader.py (hunk 1 above)
Comment (johnwlambert) [reply src/av2/datasets/sensor/av2_sensor_dataloader.py:151]: @benjaminrwilson thanks for the review. I think this logic is checking for that case already, right?

File: src/av2/datasets/sensor/av2_sensor_dataloader.py (hunk 2 above)
Comment (johnwlambert) [reply src/av2/datasets/sensor/av2_sensor_dataloader.py]: Thanks, updated now.

File: src/av2/datasets/sensor/av2_sensor_dataloader.py (hunk 1 above)
Comment (benjaminrwilson) [reply src/av2/datasets/sensor/av2_sensor_dataloader.py:151]: Yeah, for sure --- the string for the `RuntimeError` would be a little misleading in that case though, right?

Comment (johnlambert-argo) [reply src/av2/datasets/sensor/av2_sensor_dataloader.py]: Good point. I changed the check to `== 0` now, which I think is the more intuitive case (the log is not present).

**Review by wqi — APPROVED** [review]

**Pull request merged**